pkg/ipamd/ipamd.go

```diff
@@ -1060,12 +1078,24 @@ func (c *IPAMContext) setupENI(eni string, eniMetadata awsutils.ENIMetadata, isT
 				return errors.Wrapf(err, "failed to set up ENI %s network", eni)
 			}
 		}
-		log.Infof("Found ENIs having %d secondary IPs and %d Prefixes", len(eniMetadata.IPv4Addresses), len(eniMetadata.IPv4Prefixes))
-		// Either case add the IPs and prefixes to datastore.
-		c.addENIsecondaryIPsToDataStore(eniMetadata.IPv4Addresses, eni)
-		c.addENIv4prefixesToDataStore(eniMetadata.IPv4Prefixes, eni)
+		if !v6TrunkEni {
```

Comment:
`v6TrunkEni` var sounds a bit odd to me for this check here. Why can't this just be `c.enableIPv4` (or) `!c.enableIPv6` instead?

Reply:
It can be, I was trying to have a graceful solution to the case where a customer manually attaches a secondary ENI in IPv6 mode and does not mark it as unmanaged. I can change the `v6TrunkEni` checks here to just `c.enableIPv6` and comment that it currently only applies to v6 trunk ENIs